### ISSUE TITLE
Suprious rate limiting

### ISSUE THREAD

**Issue by edsu** [DocNow/twarc#445]
@sparack noticed that this little program gets rate limited almost immediately:

```python
from twarc.client2 import Twarc2

client = Twarc2(bearer_token="")
results = client.search_all(query="covid lang:en -is:retweet", max_results=10)

for tweet in results:
    print(tweet)
```

I think that the 1 request / 1 second limit mentioned for the search/all endpoint might be causing this?

https://developer.twitter.com/en/docs/twitter-api/rate-limits#v2-limits

Should Twarc2.search_all make sure that a sleep happens in order to prevent this rate limiting from happening?

**Comment by SamHames** [comment]
Oh yeah! I completely forgot about that new rate limit - definitely seems like a sleep in between calls is the right thing to do there.

**Comment by edsu** [comment]
I think if a JSON serialize and write to disk is involved that probably creates enough of a time lag, but when just pulling data over the network it's easier to run up against the 1 request / 1 second limit?

I wonder if it's possible to distinguish these rate limiting calls from the wait until the next 15 minute window?

**Comment by edsu** [comment]
I think this has been fixed in #446 which was just released as v2.0.9
